onadata/apps/api/viewsets/connect_viewset.py

```diff
@@ -135,7 +135,8 @@ def odk_token(self, request, *args, **kwargs):
         elif request.method == 'POST':
             # Regenerates the ODK Token if one is already existant
             try:
-                old_token = ODKToken.objects.get(user=user)
+                old_token = ODKToken.objects.get(
+                    user=user, status=ODKToken.ACTIVE)
```

Comment:
We do not expect it but there are chances where we may have multiple active tokens. In this scenario, we should make any active token inactive, i.e.:
```python
ODTOken.objects.filter(user=user, status=ODKToken.ACTIVE).update(status=ODKToken.INACTIVE)
```

Reply:
Added this in the latest commits. We know deactivate all tokens when two or more tokens are active then generate and return a new token.